### ISSUE TITLE
Function signature symbols missing in function type symbol of function pointer param symbols

### ISSUE THREAD

**Issue by pubudu91** [ballerina-platform/ballerina-lang#32398]
Consider something like the following
```ballerina
function foo(function(int x, float... y) returns string fn) {
}
```
When functions like the above comes from the BIR, the function signature info is missing in the function type symbol of the type of the param `fn`. Can check this by checking the function symbols of lang lib functions such as `map()`, `filter()` etc.

**Comment by sanjana** [comment]
The assumed cause for this is the params of the `tsymbol` not being read from the BIR and set appropriately. 

https://github.com/ballerina-platform/ballerina-lang/blob/a5ff9b927875c0030e7c1c40c6a03d33b65eace5/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java#L607

**Comment by sanjana** [comment]
The following cases need to be addressed. 

1. Invokable typedef (`FnType`) in a compiled source, being referred by `testFn`
```
public type FnType function(int m, int n, float p) returns string;

public function testFn(testproject:FnType fn) {
}
```

2. Function def `testFn` being defined in a compiled source.
```
public function testFn(function(int a, int b, float c) returns string fn) {
}
```

3. Invokable typedef (`FnType`) in a compiled source, consisting of recursive function pointer params (extension of case 1).
```
public type FnType function(int m, function(int a) returns string fn2) returns string;

```

3. Function pointer param consisting of other function pointer params recursively, defined in a compiled source (extension of case 2).
```
function foo(function(int x, function(int a) returns string fn2) returns string fn) {
}
```

**Comment by sanjana** [comment]
Expecting changes from #31589

**Comment by sanjana** [comment]
> @sanjana can we resume this work? Also is #37123 related to this?

@IMS94, yes. #37123 is also related to this. 

And, the Compiler front-end fix for this is expected in #36830. 
Once it is completed, need to verify in the Semantic API and add the appropriate test cases as pointed here: https://github.com/ballerina-platform/ballerina-lang/issues/32398#issuecomment-970108383

**Comment by IMS94** [comment]
Expected FE fix #37872

**Comment by IMS94** [comment]
@sanjana the FE fix had been done. We can now add the test cases and close this issue.
